generators/generatorPickerFunctions.cc

```diff
@@ -417,8 +440,21 @@ bool polynomialMassAndTPrimeSlopePicker::operator()(double& invariantMass, doubl
 		printErr << "trying to use an uninitialized massAndTPrimePicker." << endl;
 		return false;
 	}
-	TRandom3* randomNumbers = randomNumberGenerator::instance()->getGenerator();
 	invariantMass = _massPolynomial.GetRandom(_massRange.first, _massRange.second);
+	if (not pickTPrimeForMass(invariantMass, tPrime)) {
+		printErr << "error while generating t'." << std::endl;
+		return false;
+	}
+	return true;
+}
+
+
+bool polynomialMassAndTPrimeSlopePicker::pickTPrimeForMass(const double invariantMass, double& tPrime) {
```

Comment:
maybe one could improve the naming here:
`polynomialMassExponentialTPrimePicker` would probably be more appropriate, since it does not pick a slope
Not sure though, whether this would be worth breaking the user side.